### DISCUSSION TITLE
Home Ad

### DISCUSSION THREAD

**Discussion by 54m4d** [SpotX-Official/SpotX#9] [General]
Hello!

I was surprised to find this ad on the home screen, can you tell me if you also have it? is there any way to remove it?

`1.1.67.586.gbb5ef64e`

![ad](https://user-images.githubusercontent.com/52612348/132073235-7d59bb52-ed72-4081-b0b4-90a1a9c04f4d.png)

**Comment by amd64fox** [comment]
Hi, I didn't have this, you can try to catch this block, turn on developer mode, enable logging in the `config.ini` file by writing `Log = 1` in it.

The next time an advertisement appears, be sure to try to find out his CSS selector through the developer menu and write his name here in the chat, also write the log that will be in the file `blockthespot_log.txt`

I recently had an ad in a different form, the version is the same as yours:
<details>
<summary><small>Screenshot</small></summary><p>

![Снимок экрана 2021-09-01 115125](https://user-images.githubusercontent.com/62529699/132084882-badfd31c-e186-43d1-bb10-58ba81a099a8.jpg)

</details>

I also wanted to catch her, but so far she has not been there once after that incident.

**Reply by 54m4d** [reply]
I don't use blockthespot so i don't think i have `config.ini`?

<details>
<summary>Here is dev menu</summary>
<img src=https://user-images.githubusercontent.com/52612348/132091072-21b90333-8a6a-477f-8bd4-310fe684ba90.png
</details>

**Reply by amd64fox** [reply]
Hm... so my patch works in conjunction with `BlockTheSpot`, is it automatically installed,  or did you delete it after installation ? 😁

**Reply by 54m4d** [reply]
It's not the program's fault, i deleted it because i don't have ads, that's why

**Reply by amd64fox** [reply]
That is, you manually deleted BTS, I understand you correctly ?

**Reply by 54m4d** [reply]
Yes, becasue i dont have ads so no need for it, but i can install it again to try to catch this home ad if you prefer, just let me know

**Reply by amd64fox** [reply]
If you have a free account, then without BTS you will have ads, even if it is not there now, then it will appear a little later, actually as can be seen in your example.
I recommend using the BTS patch.
Or
Block it yourself via css by adding a new argument to this selector, `display: none`

**Reply by 54m4d** [reply]
I will install BTS and see i can catch it with your method above, but i really want to block it in `js` file not in `css`

**Reply by amd64fox** [reply]
If BTS blocks this ad block, I don't see any point in looking for it in the js code.

**Reply by 54m4d** [reply]
I will give you an answer...

**Reply by 54m4d** [reply]
I think the selctor is `euOnte9wvOF0D_SGxEZ9`, i found 3 matches for that selector, could you check and tell me which one i need to block?

**Reply by amd64fox** [reply]
I can't check since I don't have this ad, try adding `display:none` arguments to each selector in turn and checking the result.

**Reply by 54m4d** [reply]
This is the part of the `css` where i found the selector:
<pre>
.HiyUdbeEyoMBpq5N84ND{grid-gap:24px;display:grid;padding-top:24px}.bIdE4uyGCyx2bz0mzF4Z{-webkit-box-orient:vertical;-webkit-box-direction:normal;background:transparent;display:-webkit-box;display:-ms-flexbox;display:flex;-ms-flex-direction:column;flex-direction:column;isolation:isolate}.<b>euOnte9wvOF0D_SGxEZ9</b><b>{display:none}</b>{background-image:-webkit-gradient(linear,left top,left bottom,from(rgba(0,0,0,.6)),to(#121212)),var(--background-noise);background-image:linear-gradient(rgba(0,0,0,.6) 0,#121212 100%),var(--background-noise);height:332px;margin-top:-60px;position:absolute;-webkit-transition:background 1s ease;transition:background 1s ease;width:100%;z-index:-1}.<b>euOnte9wvOF0D_SGxEZ9</b>.rlLtWfsZLEBXHgP__yo2<b>{display:none}</b>{height:556px}.<b>euOnte9wvOF0D_SGxEZ9</b>._oiGvbZLhraKxStFAFJw<b>{display:none}</b>.IIQsrXRnN8Q4iK3ywn1J{font-size:13px!important;margin:0 16px;padding:5px 16px!important}ul .IIQsrXRnN8Q4iK3ywn1J{margin:0}
</pre>

I already added `display:none`, can you see if it is in the correct place? should i also remove the `{}`?
<br />

This is the unmodified (original) `css`:
<pre>
.HiyUdbeEyoMBpq5N84ND{grid-gap:24px;display:grid;padding-top:24px}.bIdE4uyGCyx2bz0mzF4Z{-webkit-box-orient:vertical;-webkit-box-direction:normal;background:transparent;display:-webkit-box;display:-ms-flexbox;display:flex;-ms-flex-direction:column;flex-direction:column;isolation:isolate}.<b>euOnte9wvOF0D_SGxEZ9</b>{background-image:-webkit-gradient(linear,left top,left bottom,from(rgba(0,0,0,.6)),to(#121212)),var(--background-noise);background-image:linear-gradient(rgba(0,0,0,.6) 0,#121212 100%),var(--background-noise);height:332px;margin-top:-60px;position:absolute;-webkit-transition:background 1s ease;transition:background 1s ease;width:100%;z-index:-1}.<b>euOnte9wvOF0D_SGxEZ9</b>.rlLtWfsZLEBXHgP__yo2{height:556px}.<b>euOnte9wvOF0D_SGxEZ9</b>._oiGvbZLhraKxStFAFJw{display:none}.IIQsrXRnN8Q4iK3ywn1J{font-size:13px!important;margin:0 16px;padding:5px 16px!important}ul .IIQsrXRnN8Q4iK3ywn1J{margin:0}
</pre>

**Reply by amd64fox** [reply]
```css
.euOnte9wvOF0D_SGxEZ9{display:none;background-image:...
```

**Reply by 54m4d** [reply]
Hmm, it did not work 🤔

<details>
<summary>Here is how it looks</summary>
<img src=https://user-images.githubusercontent.com/52612348/132376504-b63bebe7-c0cf-416e-aa88-26c8e9a3ef97.png>
</details>

Is it correct or there's something wrong?

**Reply by amd64fox** [reply]
I just showed you how to add an argument, and you were looking for the right selector, apparently the reason is that you chose the wrong one.

**Reply by 54m4d** [reply]
I added the argument you provided me but the add was still showing, so which one is the correct one?

**Reply by amd64fox** [reply]
Friend, I do not know which css selector is correct, I do not have this ad, so I can't tell you this, try to find it yourself.
Or just stop doing nonsense and put the bts patch in place, all advertising is blocked there.
